### PR TITLE
Initial support for charset conversion with iconv

### PR DESCRIPTION
**Pull request by Hummer12007**
Title conversion will come soon. ~~Don't merge this yet.~~

### CODE REVIEW

File: src/imap/fetch.c [hunk 1]
```diff
@@ -125,7 +126,14 @@ static void handle_body_content(struct message_part *part, imap_arg_t *args) {
 			} else if (strcasecmp(param->value, "us-ascii") == 0) {
 				// no further action necessary
 			} else {
-				worker_log(L_ERROR, "Unknown charset %s. Please report this.", param->value);
+				unsigned char *old = part->content, *new;
+				size_t news;
+				worker_log(L_DEBUG, "Converting message encoding from %s", param->value);
+				if (!(new = iconv_convert4((char *)part->content, param->value, part->size, &news))) {
```
Comment (SirCmpwn) [inline src/imap/fetch.c]: Log error

**Comment by Hummer12007** [comment]
Header conversion is broken, presumably because base64 implementation is broken, though this might not be the case.

**Comment by Hummer12007** [comment]
Yeah, worked with a different one. Still have some issues (spaces are replaced with _s in some subjects). Working on it.

**Comment by SirCmpwn** [comment]
Thanks!

**Pull request merged**